crates/transaction-pool/src/test_utils/mock.rs

```diff
@@ -1054,6 +1055,113 @@ impl TryFrom<Recovered<TransactionSigned>> for MockTransaction {
     }
 }
 
+impl TryFrom<Recovered<TxEnvelope>> for MockTransaction {
+    type Error = TryFromRecoveredTransactionError;
+
+    fn try_from(tx: Recovered<TxEnvelope>) -> Result<Self, Self::Error> {
+        let sender = tx.signer();
+        let transaction = tx.into_inner();
+        let hash = *transaction.tx_hash();
+        let size = transaction.size();
+
+        #[allow(unreachable_patterns)]
+        match transaction {
+            EthereumTxEnvelope::Legacy(signed_tx) => {
+                let tx = signed_tx.tx();
```

Comment:
you can use `.strip_signature` here and then get rid of the additional clones,

I think we should introduce `fn into_tx` in addition to strip_signature because this is more intuitive, mind submitting this to alloy as well?